Makefile

```diff
@@ -78,6 +78,60 @@ manifests: controller-gen
 generate: controller-gen
 	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
 
+# Setup development env
+# Usage: 
+# 	LAPTOP_HOST_IP=<YOUR_LAPTOP_IP> make dev-setup
+# For example:
+#	LAPTOP_HOST_IP=192.168.10.101 make dev-setup
+define tls_cnf
+[ req ]
+default_bits       = 4096
+distinguished_name = req_distinguished_name
+req_extensions     = req_ext
+[ req_distinguished_name ]
+countryName                = SG
+stateOrProvinceName        = SG
+localityName               = SG
+organizationName           = CAPSULE
+commonName                 = CAPSULE
+[ req_ext ]
+subjectAltName = @alt_names
+[alt_names]
+IP.1   = $(LAPTOP_HOST_IP)
+endef
+export tls_cnf
+dev-setup: 
+	kubectl -n capsule-system scale deployment capsule-controller-manager --replicas=0
+	mkdir -p /tmp/k8s-webhook-server/serving-certs
+	echo "$${tls_cnf}" > _tls.cnf
+	openssl req -newkey rsa:4096 -days 3650 -nodes -x509 \
+		-subj "/C=SG/ST=SG/L=SG/O=CAPSULE/CN=CAPSULE" \
+		-extensions req_ext \
+		-config _tls.cnf \
+		-keyout /tmp/k8s-webhook-server/serving-certs/tls.key \
+		-out /tmp/k8s-webhook-server/serving-certs/tls.crt
+	rm -f _tls.cnf 
+	export WEBHOOK_URL="https://$${LAPTOP_HOST_IP}:9443"; \
+	export CA_BUNDLE=`openssl base64 -in /tmp/k8s-webhook-server/serving-certs/tls.crt | tr -d '\n'`; \
+	kubectl patch MutatingWebhookConfiguration capsule-mutating-webhook-configuration \
```

Comment:
Rather than performing multiple times the kubectl patch command, couldn't we create a single patch with all the operations in it?

Reply:
sounds like a good idea but not sure what the best way is.

Just tried something like:

patch.yaml
```yaml
webhooks:
- clientConfig:
    caBundle: ${CA_BUNDLE}
    url: ${WEBHOOK_URL}/cordoning
- clientConfig:
    caBundle: ${CA_BUNDLE}
    url: ${WEBHOOK_URL}/ingresses
- clientConfig:
    caBundle: ${CA_BUNDLE}
    url: ${WEBHOOK_URL}/namespaces
- clientConfig:
    caBundle: ${CA_BUNDLE}
    url: ${WEBHOOK_URL}/networkpolicies
- clientConfig:
    caBundle: ${CA_BUNDLE}
    url: ${WEBHOOK_URL}/pods
- clientConfig:
    caBundle: ${CA_BUNDLE}
    url: ${WEBHOOK_URL}/persistentvolumeclaims
- clientConfig:
    caBundle: ${CA_BUNDLE}
    url: ${WEBHOOK_URL}/services
- clientConfig:
    caBundle: ${CA_BUNDLE}
    url: ${WEBHOOK_URL}/tenants
```

But didn't work as expected:
```sh
# Tried this
kubectl patch ValidatingWebhookConfiguration capsule-validating-webhook-configuration --type='strategic' --patch "$(envsubst < patch.yaml)"

# And this too
kubectl patch ValidatingWebhookConfiguration capsule-validating-webhook-configuration --type='merge' --patch "$(envsubst < patch.yaml)"
```

Reply:
Well, I changed a little bit and it works now:

```sh
kubectl patch ValidatingWebhookConfiguration capsule-validating-webhook-configuration \
    --type='json' -p="[\
      {'op': 'replace', 'path': '/webhooks/0/clientConfig', 'value':{'url':\"${WEBHOOK_URL}/cordoning\",'caBundle':\"${CA_BUNDLE}\"}},\
      {'op': 'replace', 'path': '/webhooks/1/clientConfig', 'value':{'url':\"${WEBHOOK_URL}/ingresses\",'caBundle':\"${CA_BUNDLE}\"}},\
      {'op': 'replace', 'path': '/webhooks/2/clientConfig', 'value':{'url':\"${WEBHOOK_URL}/namespaces\",'caBundle':\"${CA_BUNDLE}\"}},\
      {'op': 'replace', 'path': '/webhooks/3/clientConfig', 'value':{'url':\"${WEBHOOK_URL}/networkpolicies\",'caBundle':\"${CA_BUNDLE}\"}},\
      {'op': 'replace', 'path': '/webhooks/4/clientConfig', 'value':{'url':\"${WEBHOOK_URL}/pods\",'caBundle':\"${CA_BUNDLE}\"}},\
      {'op': 'replace', 'path': '/webhooks/5/clientConfig', 'value':{'url':\"${WEBHOOK_URL}/persistentvolumeclaims\",'caBundle':\"${CA_BUNDLE}\"}},\
      {'op': 'replace', 'path': '/webhooks/6/clientConfig', 'value':{'url':\"${WEBHOOK_URL}/services\",'caBundle':\"${CA_BUNDLE}\"}},\
      {'op': 'replace', 'path': '/webhooks/7/clientConfig', 'value':{'url':\"${WEBHOOK_URL}/tenants\",'caBundle':\"${CA_BUNDLE}\"}}\
    ]"
```